### ISSUE TITLE
enableReinitialize: true breaks validation

### ISSUE THREAD

**Issue by thetallweeks** [erikras/redux-form#1901]
Given a simple redux-form config like:

```
const form = reduxForm({
  form: 'ProfilesForm',
  enableReinitialize: true,
  validate
})(ProfilesForm);
```

with a validate function like:

```
const validate = values => {
  const errors = {};

  if (!values.name) {
    errors.name = 'Profile name is required.'
  }
  return errors;
};
```

the `validate` function runs twice on load and leads to an errors object:

```
{ name: 'Profile name is required.' }
```

But no errors show. What I would like to do is show an error when that field is blurred or the form is submitted. Removing `enableReinitialize: true` accomplishes this, but I need `enabledReinitialize` because the form is being populated by server data. 

Is this expected?

**Comment by itaylorweb** [comment]
I have the same problem.  

For now I have fixed by removing enableReinitialize: true and adding:

  componentWillMount() {
    **this.props.initialize();**
  }

**Comment by nicomfe** [comment]
+1

**Comment by joshbuckley182** [comment]
I'm also having this issue in 6.2. Has anyone found a solution/workaround for it?

**Comment by danielstecki** [comment]
I was able to fix this by adding an ES6 default value of an empty object to the validate function.

const validate = (values = {}) => {
  const errors = {}
  if (!values.FName) {
    errors.FName = 'Please enter a first name'
  }
.....
  return errors
}

**Comment by joshbuckley182** [comment]
@danielstecki That works for the erroneous validation call with `undefined`, but another part of the problem is when re-initialising the form with new data (e.g. switching between users on a edit screen), submitting the form succeeds when validation should fail.

**Comment by jmn8718** [comment]
@danielstecki your solution worked for me, but i am using immutable, so instead of values = {}, i used ```
```
import { fromJS } from 'immutable';
...
const validate = ( values = fromJS({})) =>
...
```

**Comment by cyuxlif** [comment]
same problem

**Comment by jplaut** [comment]
Having the same issue (also using Immutable.js). Adding a default for the values arg doesn't work either. Looking at my Redux inspector, it seems like validation is never re-running after the form reinitializes.

**Comment by dtipson** [comment]
In my case, adding enableReinitialize is actually causing the validation to run twice. The first time through it runs with no values (even though initialValues is sent down for all renders) and fails, and then the second time it's run with the values and succeeds... but the form's state is still left in an invalid state (even though making a change and then deleting it runs validation and it returns valid for all the same original values). Setting enableReinitialize to false only creates one successful validation pass again.

But my use case is I have a local record that populates a form and then when it's submit/saved, the form is sent new initialValues and so needs to switch back to "pristine" based on those values matching what's still in the fields.

**Comment by klacointe** [comment]
Same problem

**Comment by erikras** [comment]
Fix published in [`v6.4.0`](https://github.com/erikras/redux-form/releases/tag/v6.4.0).
